packages/react/select/src/Select.tsx

```diff
@@ -1010,11 +1044,31 @@ const SelectItemText = React.forwardRef<SelectItemTextElement, SelectItemTextPro
     const context = useSelectContext(ITEM_TEXT_NAME, __scopeSelect);
     const contentContext = useSelectContentContext(ITEM_TEXT_NAME, __scopeSelect);
     const itemContext = useSelectItemContext(ITEM_TEXT_NAME, __scopeSelect);
-    const ref = React.useRef<SelectItemTextElement | null>(null);
-    const composedRefs = useComposedRefs(forwardedRef, ref, itemContext.onItemTextChange, (node) =>
-      contentContext.itemTextRefCallback?.(node, itemContext.value, itemContext.disabled)
+    const itemsContext = useSelectNativeOptionsContext(ITEM_TEXT_NAME, __scopeSelect);
+    const [itemTextNode, setItemTextNode] = React.useState<SelectItemTextElement | null>(null);
+    const composedRefs = useComposedRefs(
+      forwardedRef,
+      (node) => setItemTextNode(node),
+      itemContext.onItemTextChange,
+      (node) => contentContext.itemTextRefCallback?.(node, itemContext.value, itemContext.disabled)
+    );
+
+    const textContent = itemTextNode?.textContent;
+    const nativeOption = React.useMemo(
+      () => (
+        <option key={itemContext.value} value={itemContext.value} disabled={itemContext.disabled}>
+          {textContent}
+        </option>
+      ),
+      [itemContext.disabled, itemContext.value, textContent]
     );
```

Comment:
I am building the option element directly here, this has 2 advantages in my opinion:
- it makes it more clear/explicit what this is used for, seeing the native option collocated here
- it means I don't need to invent a new data structure like `{ value, label: textContent, disabled }` just to pass it up and interpolate it higher up.

Reply:
haha yeah.. my initial gut reaction was to ask why we don't render closer, I think I'm just conditioned by typically seeing these built closer to the render. But:

> it means I don't need to invent a new data structure like { value, label: textContent, disabled } just to pass it up and interpolate it higher up.

Actually makes much more sense, it's easier to follow when you realise the alternative is just going around the houses for no great reason.